File: src/backend/schemas/spend.py
from decimal import Decimal

from pydantic import BaseModel


class CategorySpendRow(BaseModel):
    category_id: int
    name: str
    actual: Decimal
    budget: Decimal | None


class SpendSummary(BaseModel):
    rows: list[CategorySpendRow]
    uncategorised_actual: Decimal
    total_actual: Decimal
    total_budget: Decimal | None
    has_transactions: bool

File: src/frontend/src/lib/api/spend.ts
export interface CategorySpendRow {
  category_id: number;
  name: string;
  actual: string;
  budget: string | null;
}

export interface SpendSummary {
  rows: CategorySpendRow[];
  uncategorised_actual: string;
  total_actual: string;
  total_budget: string | null;
  has_transactions: boolean;
}

export async function getSpendSummary(): Promise<SpendSummary> {
  const res = await fetch('/api/spend/summary');
  if (!res.ok) throw new Error('Failed to load spend summary');
  return res.json();
}
